p2p/discovery.py

```diff
@@ -419,15 +569,15 @@ def _pack(cmd_id: int, payload: List[Any], privkey: datatypes.PrivateKey) -> byt
     how UDP packets are structured.
     """
     cmd_id = to_bytes(cmd_id)
-    expiration = rlp.sedes.big_endian_int.serialize(int(time.time() + EXPIRATION))
+    expiration = rlp.sedes.big_endian_int.serialize(_get_msg_expiration())
     encoded_data = cmd_id + rlp.encode(payload + [expiration])
     signature = privkey.sign_msg(encoded_data)
     message_hash = keccak(signature.to_bytes() + encoded_data)
     return message_hash + signature.to_bytes() + encoded_data
 
 
-def _unpack(message: bytes) -> Tuple[datatypes.PublicKey, int, List[Any], bytes]:
-    """Unpack a UDP message received from a remote node.
+def _unpack(message: bytes) -> Tuple[datatypes.PublicKey, int, List[Any], Hash32]:
```

Comment:
Maybe go ahead and include the `_v4` suffix for these so that there's no ambiguity or accidental mis-use of v4/v5?  We can easily drop the suffixes once we're fully onto v5.